### PR TITLE
Alwallace/saiuaiupdate

### PR DESCRIPTION
**Pull request by xanwal**
# Description

- Add an SDK-based way to retrieve blobs in scoring scripts

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/Azure/azureml-examples/blob/main/CONTRIBUTING.md)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by shohei1029 — APPROVED** [review]
Thanks for the outstanding PR. It looks good to me!
I understand that the PR was tested by Dan.

File: cli/endpoints/online/model-1/onlinescoring/score_managedidentity.py [hunk 1]
```diff
@@ -4,9 +4,31 @@
 import numpy
 import joblib
 import requests
+from azure.identity import ManagedIdentityCredential
+from azure.storage.blob import BlobClient
 
 
-def get_token():
+def access_blob_storage_sdk():
+    credential = ManagedIdentityCredential()
```
Comment (tomasvanpottelbergh) [inline cli/endpoints/online/model-1/onlinescoring/score_managedidentity.py:12]: I don't think this works for user-assigned managed identities, think you'd need something like this:
```suggestion
    client_id = os.getenv('UAI_CLIENT_ID')
    credential = ManagedIdentityCredential(client_id=client_id)
```

File: sdk/python/endpoints/online/model-1/onlinescoring/score_managedidentity.py [hunk 2]
```diff
@@ -4,10 +4,31 @@
...
 import joblib
 import requests
-from pathlib import Path
+from azure.identity import ManagedIdentityCredential
+from azure.storage.blob import BlobClient
 
 
-def get_token():
+def access_blob_storage_sdk():
+    credential = ManagedIdentityCredential()
```
Comment (tomasvanpottelbergh) [inline sdk/python/endpoints/online/model-1/onlinescoring/score_managedidentity.py:12]: Same here, I think this should be
```suggestion
    client_id = os.getenv('UAI_CLIENT_ID')
    credential = ManagedIdentityCredential(client_id=client_id)
```

**Review by tomasvanpottelbergh — CHANGES_REQUESTED** [review]

**Comment by xanwal** [comment]
@DannyBruno Can you e2e test this as part of your review in the sandbox workspace?

**Review by tomasvanpottelbergh — APPROVED** [review]

**Review by dem108 — APPROVED** [review]
Thanks! It will serve as a great baseline.

**Pull request merged**